app/lib/import_export.rb

```diff
@@ -652,8 +654,7 @@ def extract_incomplete_patient_details(patient, fields)
 
   # Gets a hash of the latest transfers of each patient
   def get_patients_transfers(patients)
-    transfers = patients.pluck(:id, :latest_transfer_at)
-    transfers = Transfer.where(patient_id: transfers.map { |lt| lt[0] }, created_at: transfers.map { |lt| lt[1] })
+    transfers = Transfer.where(patient_id: patients.pluck(:id), created_at: patients.pluck(:latest_transfer_at))
```

Comment:
Would it be possible for the `created_at` to be the same for two Transfers for two different patients? If that is the case then could we expect the possibility of two or more transfers being returned for one patient when we only want one per patient?

The SQL generated from `Transfer.where(patient_id: [...], created_at: [...])` is 
```sql
SELECT `transfers`.* FROM `transfers` WHERE `transfers`.`patient_id` IN (...) AND `transfers`.`created_at` IN (...) LIMIT N
``` 

| patient_id | created_at                           |
| ---------- | ------------------------------------ |
| 1          | Fri, 25 Dec 2020 19:02:04 UTC +00:00 |
| 2          | Fri, 25 Dec 2020 19:02:04 UTC +00:00 |
| 2          | Sat, 26 Dec 2020 20:17:20 UTC +00:00 |

Patient 1's latest_transfer_at is on Friday and patient 2's latest_transfer_at is on Saturday, but the way the SQL is generated we could also expect patient 2's transfer on Friday to be returned even though it is not that patient's latest transfer.

Reply:
So two things:
1. Separate from your point here, I'm actually going to revert the code change I made to this line because it was meant to be a slight optimization but is technically different functionality if pluck does not return the values in the same order. 
2. I think you're right that this is problematic for the case you're outlining, though I think it's probably pretty rare if it's ever happened at all just because it would need to be exact same second. For the sake of minimizing any functionality changes in this patch which is entirely meant to resolve memory issues, I'm going to instead make a separate ticket to resolve this edge case and put at the top of our backlog. Great catch!